migrate workspace to turborepo

alecananian opened the pull request:
- Migrates the workspace to use turborepo for main commands
- Fixes various typecheck issues in example apps

nickfujita commented:
Not sure if this is related to turbo, but when running `dev:react` or `start:api` the api fails to start with the following error:
```
import { PrismaClient } from "@prisma/client";
         ^^^^^^^^^^^^
SyntaxError: Named export 'PrismaClient' not found. The requested module '@prisma/client' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@prisma/client';
const { PrismaClient } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:134:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:217:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:123:5)
    ```

nickfujita commented:
Overall flow looks good. Just a couple notes:
- Not a big deal, but when running `dev:react` the dev process for launcher package is running. is this needed there?
- for ci if env variables are set in the workflows when calling a turbo process, please make sure to configure those env variables to be passed in `turbo.json`. You can set these on a per task basis, or globally in `globalEnv`. [For example](https://github.com/TreasureProject/flame/blob/main/turbo.json#L4).

alecananian commented:
Thanks for taking a look! Addressing the comments below:
- Fixed the `PrismaClient` by adding the `generate` step as a dependency on `dev` commands
- Yes the React package relies on exports from the Launcher package
- I don't think we need to pass any envs to the CI, but I'll find out shortly if it fails on merge 🙂
